diff -urN mtr-0.85/select.c mtr-0.85-patched/select.c
--- mtr-0.85/select.c	2014-03-19 04:02:12.217729857 +1100
+++ mtr-0.85-patched/select.c	2014-03-19 04:03:12.606753119 +1100
@@ -58,6 +58,7 @@
 #endif
   int NumPing = 0;
   int paused = 0;
+  int quit = 0;
   struct timeval lasttime, thistime, selecttime;
   int dt;
   int rv; 
@@ -117,9 +118,10 @@
 	   (thistime.tv_sec == lasttime.tv_sec + intervaltime.tv_sec &&
 	    thistime.tv_usec >= lasttime.tv_usec + intervaltime.tv_usec)) {
 	  lasttime = thistime;
+	  if(quit == 1) return;
 	  if(NumPing >= MaxPing && (!Interactive || ForceMaxPing))
-	    return;
-	  if (net_send_batch())
+	    quit = 1;
+	  else if (net_send_batch())
 	    NumPing++;
 	}
 
