diff -urN mtr-0.85/report.c mtr-0.85-patched/report.c
--- mtr-0.85/report.c	2014-03-19 04:02:12.218729874 +1100
+++ mtr-0.85-patched/report.c	2014-03-19 04:02:25.411953426 +1100
@@ -294,7 +294,7 @@
     printf("    <HUB COUNT=\"%d\" HOST=\"%s\">\n", at+1, name);
     for( i=0; i<MAXFLD; i++ ) {
       j = fld_index[fld_active[i]];
-      if (j < 0) continue;
+      if (j <= 0) continue;
 
       strcpy(name, "        <%s>");
       strcat(name, data_fields[j].format);
